vmm/src/memory_manager.rs

```diff
@@ -228,8 +241,31 @@ impl MemoryManager {
             mem_end.unchecked_add(1)
         };
 
+        let mut virtiomem_region = None;
+        let mut virtiomem_resize = None;
         if let Some(size) = hotplug_size {
-            start_of_device_area = start_of_device_area.unchecked_add(size);
+            if hotplug_method == HotplugMethod::VirtioMem {
+                let start_addr = if start_of_device_area < arch::layout::MEM_32BIT_RESERVED_START {
+                    arch::layout::RAM_64BIT_START
```

Comment:
since start_of_device_area is alwasy >= arch::layout::MEM_32BIT_RESERVED_START

we may remove the code here to make it simple.

Reply:
> since start_of_device_area is alwasy >= arch::layout::MEM_32BIT_RESERVED_START
> 
> we may remove the code here to make it simple.

Removed.